router/router.go

```diff
@@ -111,14 +113,18 @@ func New(cfg Config) http.Handler {
 
 	if cfg.SearchRoutesEnabled {
 		if cfg.EnableSearchABTest {
-			router.Handle("/search", abtest.SearchHandler(cfg.SearchHandler, cfg.BabbageHandler, cfg.SearchABTestPercentage, cfg.SiteDomain))
+			router.Handle("/search", abtest.ABTestHandler(cfg.SearchHandler, cfg.BabbageHandler, cfg.SearchABTestPercentage, abtest.SearchTestCookieAspect, cfg.SiteDomain, abtest.SearchNewExit))
 		} else {
-			router.Handle("/search", cfg.SearchHandler)
+			router.Handle("/search", abtest.ABTestPurgeHandler(cfg.SearchHandler, abtest.SearchTestCookieAspect, cfg.SiteDomain))
 		}
 	}
 
 	if cfg.RelCalEnabled {
-		router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", cfg.RelCalHandler)
+		if cfg.RelCalEnableABTest {
+			router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", abtest.ABTestHandler(cfg.RelCalHandler, cfg.BabbageHandler, cfg.RelCalABTestPercentage, abtest.RelcalTestCookieAspect, cfg.SiteDomain, abtest.RelcalNewExit))
+		} else {
+			router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", abtest.ABTestPurgeHandler(cfg.RelCalHandler, abtest.RelcalTestCookieAspect, cfg.SiteDomain))
```

Comment:
May I suggest having a single function in `abtest` that returns the relevant handler based on the flag value? It could avoid duplicating code every time we need ab testing for a specific route?